### PR TITLE
Unit Commitment Fixes

### PR DESCRIPTION
**Pull request by sourabhdalvi**
- Bug fixed in the Cost function
- Look-ahead implementation of minimum-up/down time constraint
- Parameter JuMP Compatible implementation of Minimum-UP/Down time constraints

### CODE REVIEW

**Comment by jd-lara** [comment]
@sourabhdalvi could you please resolve the conflicts?

**Comment by sourabhdalvi** [comment]
Working on it

File: src/devices/device_models/common/duration_constraints.jl [hunk 1]
```diff
@@ -19,30 +19,36 @@ function device_duration_retrospective(ps_m::CanonicalModel,
...
+            end
+            (t - round(tst) + 1) < 0 ? JuMP.add_to_expression!(lhs_on,1) : nothing
+            
+            ps_m.constraints[name_up][name, t] = JuMP.@constraint(ps_m.JuMPmodel, 
+                lhs_on <= ps_m.variables[var_names[1]][name,t])
+            
+            # Minimum Down-time Constraint
+            lhs_off = JuMP.GenericAffExpr{Float64, _variable_type(ps_m)}(0);
+            for i in ((t - round(tsd) + 1) :t)
+                in(time_steps,i) ? JuMP.add_to_expression!(lhs_off,1,ps_m.variables[var_names[3]][name,i]) : nothing ;
```
Comment (jd-lara) [inline src/devices/device_models/common/duration_constraints.jl]: Can we implement this if-else statements?

File: src/devices/device_models/common/duration_constraints.jl [hunk 2]
```diff
@@ -124,16 +130,24 @@ function device_duration_look_ahead(ps_m::CanonicalModel,
...
         ps_m.constraints[name_up][name, t] = JuMP.@constraint(ps_m.JuMPmodel, 
-            sum([ps_m.variables[var_names[1]][name,i] for i in t-duration_data[ix].up:t if in(time_steps,i)])
-            + duration_ind_status_on[ix].value
-            <= ps_m.variables[var_names[3]][name,t])*duration_data[ix].up
+            lhs_on <= ps_m.variables[var_names[3]][name,t])*duration_data[ix].up
 
         # Minimum Down-time Constraint
+        lhs_off = JuMP.GenericAffExpr{Float64, _variable_type(ps_m)}(0);
+        for i in t-duration_data[ix].up:t
+            in(time_steps,i) ? JuMP.add_to_expression!(lhs_off,(1-ps_m.variables[var_names[1]][name,i])) : nothing ;
```
Comment (jd-lara) [inline src/devices/device_models/common/duration_constraints.jl]: Same here, let's use if-else statements.

**Comment by sourabhdalvi** [comment]
I made the requested changes, removed merge conflicts. I still need to test the constraints more rigorously for now they seem to work. Let me know if you need anything else ?

**Pull request merged**